pkg/ovs/openflow/ofctrl_packetout_test.go

```diff
@@ -20,11 +20,263 @@ import (
+		pb := b.SetICMPData(data)
+		assert.Equal(t, data, pb.(*ofPacketOutBuilder).pktOut.ICMPHeader.Data)
+	})
+	t.Run("SetUDPData", func(t *testing.T) {
+		data := []byte{0x11, 0x22}
+		b := newPktOutBuilder()
+		pb := b.SetUDPData(data)
+		assert.Equal(t, data, pb.(*ofPacketOutBuilder).pktOut.UDPHeader.Data)
+	})
+	t.Run("SetInport", func(t *testing.T) {
+		inPort := uint32(1)
+		b := newPktOutBuilder()
+		pb := b.SetInport(inPort)
+		assert.Equal(t, inPort, pb.(*ofPacketOutBuilder).pktOut.InPort)
+	})
+	t.Run("SetOutport", func(t *testing.T) {
+		outPort := uint32(1)
+		b := newPktOutBuilder()
+		pb := b.SetOutport(outPort)
+		assert.Equal(t, outPort, pb.(*ofPacketOutBuilder).pktOut.OutPort)
+	})
+	t.Run("SetL4Packet", func(t *testing.T) {
+		igmp := &protocol.IGMPv1or2{
+			Type:         protocol.IGMPQuery,
+			GroupAddress: net.ParseIP("1.2.3.4"),
+		}
+		b := newPktOutBuilder()
+		b.pktOut.IPHeader = new(protocol.IPv4)
+		pb := b.SetL4Packet(igmp)
+		assert.Equal(t, igmp, pb.(*ofPacketOutBuilder).pktOut.IPHeader.Data)
+	})
+	t.Run("AddSetIPTOSAction", func(t *testing.T) {
+		data := uint8(1)
+		b := newPktOutBuilder()
+		pb := b.AddSetIPTOSAction(data)
+		pktOut := pb.(*ofPacketOutBuilder).pktOut
+
+		assert.Equal(t, 1, len(pktOut.Actions))
+		action := pktOut.Actions[0]
+
+		assert.IsType(t, &ofctrl.SetFieldAction{}, action)
+		setFieldAction := action.(*ofctrl.SetFieldAction)
+
+		assert.Equal(t, openflow15.OXM_CLASS_NXM_0, int(setFieldAction.Field.Class))
+		assert.Equal(t, openflow15.NXM_OF_IP_TOS, setFieldAction.Field.Field)
+
+		assert.IsType(t, &openflow15.IpDscpField{}, setFieldAction.Field.Value)
+		value := setFieldAction.Field.Value.(*openflow15.IpDscpField)
+		assert.Equal(t, data<<2, value.Dscp)
+	})
+	t.Run("AddLoadRegMark", func(t *testing.T) {
+		mark := NewRegMark(NewRegField(1, 0, 15), 0x1234)
+		b := newPktOutBuilder()
+		pb := b.AddLoadRegMark(mark)
+		pktOut := pb.(*ofPacketOutBuilder).pktOut
+
+		assert.Equal(t, 1, len(pktOut.Actions))
+		action := pktOut.Actions[0]
+
+		assert.IsType(t, &ofctrl.SetFieldAction{}, action)
+		setFieldAction := action.(*ofctrl.SetFieldAction)
+
+		assert.Equal(t, openflow15.OXM_CLASS_NXM_1, int(setFieldAction.Field.Class))
+		assert.Equal(t, openflow15.NXM_NX_REG1, int(setFieldAction.Field.Field))
+
+		assert.IsType(t, &openflow15.Uint32Message{}, setFieldAction.Field.Value)
+		assert.IsType(t, &openflow15.Uint32Message{}, setFieldAction.Field.Mask)
+		value := setFieldAction.Field.Value.(*openflow15.Uint32Message)
+		mask := setFieldAction.Field.Mask.(*openflow15.Uint32Message)
+
+		assert.Equal(t, uint32(0x1234), value.Data)
+		assert.Equal(t, uint32(0xffff), mask.Data)
+	})
+	t.Run("AddResubmitAction", func(t *testing.T) {
+		table := uint8(1)
+		b := newPktOutBuilder()
+		pb := b.AddResubmitAction(nil, &table)
+		pktOut := pb.(*ofPacketOutBuilder).pktOut
+
+		assert.Equal(t, 1, len(pktOut.Actions))
+		action := pktOut.Actions[0]
+
+		assert.IsType(t, &ofctrl.Resubmit{}, action)
+	})
```

Comment:
Missing the validation on the resubmitted table?

Reply:
The field in action is private then we cannot validate it.